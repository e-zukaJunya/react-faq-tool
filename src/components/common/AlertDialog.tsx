import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import React from "react";

type Props = {
  open: boolean;
  onClickClose: () => void;
};

const AlertDialog: React.FC<Props> = props => {
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div className="">
      <Button variant="outlined" color="primary" onClick={handleClickOpen} className={""}>
        Open alert dialog
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{"Use Google's location service?"}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">確認ダイアログ</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary" autoFocus>
            閉じる
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default AlertDialog;
