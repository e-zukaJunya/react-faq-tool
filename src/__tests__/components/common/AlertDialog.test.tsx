import React from "react";
import { createShallow, createMount } from "@material-ui/core/test-utils";
import { shallow, mount } from "enzyme";
import AlertDialog from "components/common/AlertDialog";
import renderer from "react-test-renderer";
import {createRenderer} from 'react-test-renderer/shallow';

describe("AlertDialog", () => {
  //   jest.mock("../SomeDirectory/SomeComponent", () => "SomeComponent");

  beforeEach(() => {});

  test("open", async () => {
    const testRenderer =  createRenderer();
    // const renderer =  ShallowRenderer();
    const wrapper = testRenderer.render(<AlertDialog open={true} onClickClose={() => {}} />);
    const wrapper2 = renderer.create(<AlertDialog open={true} onClickClose={() => {}} />);
    // const root =  wrapper2.root
    expect(wrapper).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    // expect(root).toMatchSnapshot();
    // expect(wrapper.toJSON()).toMatchSnapshot();
    // const wrapper2 = mount(<AlertDialog open={true} onClickClose={() => {}} />);
    const wrapper3 = shallow(<AlertDialog open={true} onClickClose={() => {}} />);
    // wrapper2.children()
    const button = wrapper3.find("button");
    // console.log(button.html())
    console.log(button.length);
    // wrapper2.children()
    // wrapper2.shallow()
    // wrapper2.update()
    // wrapper2.dive()
    // console.log(button.html())
    const children = button.children();
    const childrenButton = button.children("button");
    const parent = wrapper3.parent();
    const parents = wrapper3.parents();
    // const children2 = wrapper3.dive().children("button");
    const children3 = wrapper3.parent().find("button");
    const children4 = wrapper3.parents().find("button");
    const children5 = wrapper3.parents().children().find("button");
    console.log(children.length);
    // console.log(children2.length);
    console.log(parent.length);
    console.log(parents.length);
    console.log(children3.length);
    console.log(children4.length);
    console.log(children5.length);
    console.log(childrenButton.length);
    // expect(children2).toMatchSnapshot();
  });

//   test("close", async () => {
//     const wrapper = renderer.create(<AlertDialog open={false} onClickClose={() => {}} />);
//     expect(wrapper.toJSON()).toMatchSnapshot();
//   });
});
