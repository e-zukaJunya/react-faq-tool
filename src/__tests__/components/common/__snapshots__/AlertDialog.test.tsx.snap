// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog open 1`] = `
<div
  className=""
>
  <ForwardRef(WithStyles)
    className=""
    color="primary"
    onClick={[Function]}
    variant="outlined"
  >
    Open alert dialog
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(WithStyles)
      id="alert-dialog-title"
    >
      Use Google's location service?
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        id="alert-dialog-description"
      >
        アラートダイアログ
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        autoFocus={true}
        color="primary"
        onClick={[Function]}
      >
        閉じる
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;

exports[`AlertDialog open 2`] = `
<div
  className=""
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Open alert dialog
    </span>
  </button>
</div>
`;
